scannerc.go

```diff
@@ -859,6 +864,20 @@ func yaml_simple_key_is_valid(parser *yaml_parser_t, simple_key *yaml_simple_key
 	return true, true
 }
 
+func yaml_parser_stale_simple_keys(parser *yaml_parser_t) bool {
+	// fmt.Println(parser.simple_keys_min_possible_index, ":", len(parser.simple_keys))
+	for i := parser.simple_keys_min_possible_index; i < len(parser.simple_keys); i++ {
```

Comment:
I'd like to avoid adding this linear search back. I understand why it existed and why it's being suggested again, but in principle another linear search is worse than fixing the same issue without having it. But let's discuss the earlier issue first and then I'll think about this case some more.

Reply:
Ack. I think this is unnecessary if we can always count on checking the token at the head of the parser for staleness.